docs/modules/ROOT/pages/erc721-consecutive.adoc

```diff
@@ -0,0 +1,166 @@
+= ERC-721 Consecutive
+
+Consecutive extension for xref:erc721.adoc[ERC-721] is useful for efficiently minting multiple tokens in a single transaction. This can significantly reduce gas costs and improve performance when creating a large number of tokens at once.
+
+[[usage]]
+== Usage
+
+In order to make https://docs.rs/openzeppelin_stylus/token/erc721/extensions/consecutive/index.html[`ERC-721 Consecutive`] methods “external” so that other contracts can call them, you need to add the following code to your contract:
+
+[source,rust]
+----
+use openzeppelin_stylus::token::erc721::extensions::consecutive::{
+    Erc721Consecutive, Error,
+};
+
+sol_storage! {
+    #[entrypoint]
+    struct Erc721ConsecutiveExample {
+        #[borrow]
+        Erc721Consecutive erc721_consecutive;
+    }
+}
+
+#[external]
+#[inherit(Erc721Consecutive)]
+impl Erc721ConsecutiveExample {
+    pub fn burn(&mut self, token_id: U256) -> Result<(), Error> {
+        self.erc721_consecutive._burn(token_id)
+    }
+
+    pub fn mint(&mut self, to: Address, token_id: U256) -> Result<(), Error> {
+        self.erc721_consecutive._mint(to, token_id)
+    }
+}
+----
+
+Additionally, you need to ensure proper initialization during xref:deploy.adoc[contract deployment]. Make sure to include the following code in your Solidity Constructor:
+
+[source,solidity]
+----
+contract Erc721ConsecutiveExample {
+    mapping(uint256 tokenId => address) private _owners;
+    mapping(address owner => uint256) private _balances;
+    mapping(uint256 tokenId => address) private _tokenApprovals;
+    mapping(address owner => mapping(address operator => bool)) private _operatorApprovals;
+
+    Checkpoint160[] private _checkpoints; // _sequentialOwnership
+    mapping(uint256 bucket => uint256) private _data; // _sequentialBurn
+    uint96 private _firstConsecutiveId;
+    uint96 private _maxBatchSize;
+
+    error ERC721InvalidReceiver(address receiver);
+    error ERC721ForbiddenBatchMint();
+    error ERC721ExceededMaxBatchMint(uint256 batchSize, uint256 maxBatch);
+    error ERC721ForbiddenMint();
+    error ERC721ForbiddenBatchBurn();
+    error CheckpointUnorderedInsertion();
+
+    event ConsecutiveTransfer(
+        uint256 indexed fromTokenId,
+        uint256 toTokenId,
+        address indexed fromAddress,
+        address indexed toAddress
+    );
+
+    struct Checkpoint160 {
+        uint96 _key;
+        uint160 _value;
+    }
+
+    constructor(
+        address[] memory receivers,
+        uint96[] memory amounts,
+        uint96 firstConsecutiveId,
+        uint96 maxBatchSize)
```

Comment:
That looks a bit cringe. But we need to have that big constructor for `Consecutive` with `koba`